File: tests/adapters/javascript/goldens/imports/strip_all.js
/**
 * JavaScript module for testing import optimization.
 */

// … 72 imports omitted (62 lines)

// Dynamic imports (should be preserved as-is)
const dynamicModule = async () => {
    const { default: chalk } = await import('chalk');
    return chalk;
};

// Conditional imports
let csvParser;
try {
    csvParser = require('csv-parser');
} catch (error) {
    console.warn('csv-parser not available');
}

export class ImportTestService {
    constructor(userService, dbConnection, logger) {
        this.userService = userService;
        this.dbConnection = dbConnection;
        this.logger = logger;
    }

    async processData(data) {
        // Using external libraries
        const processed = lodash.map(data, item => ({
            id: uuidv4(),
            timestamp: moment().toISOString(),
            ...item
        }));

        // Using local utilities
        const validated = processed.map(item =>
            validateEmail(item.email) ? item : null
        ).filter(Boolean);

        // Using Node.js built-ins
        const filePath = path.join(__dirname, 'output.json');
        await promisify(fs.writeFile)(filePath, JSON.stringify(validated));

        return {
            success: true,
            data: validated,
            timestamp: formatDate(new Date())
        };
    }

    async makeHttpRequest(url) {
        try {
            // Using axios
            const response = await axios.get(url, {
                timeout: 5000,
                headers: {
                    'User-Agent': 'ImportTestService/1.0'
                }
            });

            return response.data;
        } catch (error) {
            this.logger.error('HTTP request failed', error);
            throw new NetworkError('Request failed');
        }
    }
}

// … import omitted (4 lines)

// Default export
export default ImportTestService;

File: tests/adapters/scala/goldens/imports/strip_all.scala
/**
 * Scala module for testing import optimization.
 */

package com.example.imports

// … 90 imports omitted (81 lines)

class ImportTestService(
  userService: UserService,
  dbConnection: DatabaseConnection,
  logger: AppLogger
)(implicit ec: ExecutionContext) {

  def processData(data: List[Any]): Future[ApiResponse[List[Any]]] = {
    // Using external libraries
    val processed = data.map { item =>
      Map(
        "id" -> UUID.randomUUID(),
        "timestamp" -> Instant.now().toString,
        "item" -> item
      )
    }

    // Using local utilities
    val validated = processed.filter { item =>
      item.get("email")
        .map(_.toString)
        .exists(validateEmail)
    }

    // Using Scala standard library
    val filePath = "output.json"

    Future.successful(
      ApiResponse(
        success = true,
        data = validated,
        timestamp = formatDate(LocalDateTime.now())
      )
    )
  }

  def makeHttpRequest(url: String)(implicit system: ActorSystem): Future[String] = {
    // … import omitted

    // Using Akka HTTP
    val request = HttpRequest(uri = url)
      .withHeaders(
        headers.`User-Agent`("ImportTestService/1.0")
      )

    Http()
      .singleRequest(request)
      .flatMap { response =>
        Unmarshal(response.entity).to[String]
      }
      .recover {
        case e: Exception =>
          logger.error("HTTP request failed", e)
          throw new NetworkError("Request failed")
      }
  }

  def serializeData(data: Any): String = {
    // Using Play JSON
    val json = Json.toJson(data)
    Json.prettyPrint(json)
  }
}

// Case class with validation annotations
case class ValidatedUser(
  id: String,
  name: String,
  email: String,
  age: Option[Int],
  phone: Option[String]
)

// Akka HTTP route
trait UserRoutes {
  implicit def system: ActorSystem
  implicit def executionContext: ExecutionContext

  val userRoute: Route =
    pathPrefix("api" / "users") {
      concat(
        path(LongNumber) { id =>
          get {
            complete(s"Get user $id")
          }
        },
        pathEnd {
          post {
            entity(as[ValidatedUser]) { user =>
              complete(StatusCodes.Created, user)
            }
          }
        }
      )
    }
}

// Play JSON format
object UserFormats {
  implicit val userFormat: Format[ValidatedUser] = Json.format[ValidatedUser]
}

// Slick database query
class UserTable(tag: Tag) extends Table[ValidatedUser](tag, "users") {
  def id = column[String]("id", O.PrimaryKey)
  def name = column[String]("name")
  def email = column[String]("email")
  def age = column[Option[Int]]("age")
  def phone = column[Option[String]]("phone")

  def * = (id, name, email, age, phone).mapTo[ValidatedUser]
}

// Cats Effect IO
object CatsEffectExample {
  def processWithIO(items: List[String]): IO[List[String]] =
    items.traverse { item =>
      IO.sleep(100.millis) *>
        IO.pure(item.toUpperCase)
    }
}

// Default export
object Main extends App {
  implicit val system: ActorSystem = ActorSystem("ImportTestSystem")
  implicit val ec: ExecutionContext = system.dispatcher

  val config = ConfigFactory.load()
  println(s"Application started with config: ${config.getString("app.name")}")
}
